tests/phpunit/tests/script-modules/wpScriptModules.php

```diff
@@ -1728,4 +1732,89 @@ public static function data_invalid_script_module_data(): array {
 			'string'   => array( 'string' ),
 		);
 	}
+
+	/**
+	 * Tests that load_on_client_navigation defaults to false when registering a script module.
+	 *
+	 * @covers WP_Script_Modules::register
+	 */
+	public function test_load_on_client_navigation_defaults_to_false_for_registered_modules() {
+		$this->script_modules->register( 'foo', '/foo.js' );
+		$registered_modules = $this->get_registered_script_modules( $this->script_modules );
+		$this->assertSame( false, $registered_modules['foo']['load_on_client_navigation'] );
```

Comment:
More nitpicking. 😅 You could use `assertFalse` here instead of `assertSame( false, ... )`. Same in other tests below with `assertTrue`.
```suggestion
		$this->assertFalse( $registered_modules['foo']['load_on_client_navigation'] );
```